ReactiveCocoaTests/Swift/SignalLifetimeSpec.swift

```diff
@@ -210,6 +210,91 @@ class SignalLifetimeSpec: QuickSpec {
 				expect(signal).to(beNil())
 			}
 		}
+
+		describe("observe") {
+			/// Used to detect the lifetime of `Signal` handlers.
+			final class Token {
+				private let deallocHandler: () -> ()
+
+				init(deallocHandler: () -> ()) {
+					self.deallocHandler = deallocHandler
+				}
+
+				func test() {}
+
+				deinit { self.deallocHandler() }
+			}
+
+
+			var signal: Signal<Int, TestError>!
+			var observer: Signal<Int, TestError>.Observer!
+
+			var tokenWasDeallocated: Bool!
+
+			var token: Token? = nil
+			weak var weakToken: Token?
+
+			func expectTokenNotDeallocated() {
+				expect(tokenWasDeallocated) == false
+				expect(weakToken).toNot(beNil())
```

Comment:
Yeah I guess you're right. I wanted to be as exhaustive as possible but I can't think of any reason why just relying on `weak` wouldn't be enough.